Client/FrameBuffer.cpp

```diff
@@ -0,0 +1,124 @@
+﻿#include "FrameBuffer.h"
+#include <glad/glad.h>
+#include <glm/glm.hpp>
+#include <memory>
+#include "Shader.hpp"
+
+static const GLfloat g_quad_vertex_buffer_data[] = {
+  -1.0f, -1.0f, 0.0f,
+  1.0f, -1.0f, 0.0f,
+  -1.0f, 1.0f, 0.0f,
+  -1.0f, 1.0f, 0.0f,
+  1.0f, -1.0f, 0.0f,
+  1.0f, 1.0f, 0.0f,
+};
+
+FrameBuffer::FrameBuffer() : FrameBuffer(1024, 1024) {
+
+}
+
+FrameBuffer::FrameBuffer(int width, int height) {
+  this->width = width;
+  this->height = height;
+  glGenFramebuffers(1, &frameBufferID);
+  glBindFramebuffer(GL_FRAMEBUFFER, frameBufferID);
+
+  glGenTextures(1, &rgbTexture);
+
+  // "Bind" the newly created texture : all future texture functions will modify this texture
+  glBindTexture(GL_TEXTURE_2D, rgbTexture);
+
+  // Give an empty image to OpenGL ( the last "0" )
+  glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA, this->width, this->height, 0, GL_RGBA, GL_UNSIGNED_BYTE, 0);
+
+  // Poor filtering. Needed !
+  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
+  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_BORDER);
+  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_BORDER);
+  float color[] = {0.0f, 0.0f, 0.0f, 0.0f};
+  glTexParameterfv(GL_TEXTURE_2D, GL_TEXTURE_BORDER_COLOR, color);
+
+  glGenRenderbuffers(1, &depthBufferID);
```

Comment:
The one thing I think might be valuable is having a helper function just for creating a depth texture and on for creating a depth buffer. Since it might not always be the case that we need both.

Reply:
This might be a good idea, I think we may discuss more about it tomorrow

Reply:
This is because adding a depth buffer and depth texture only cause 1/a little overhead. 
2/Breaking it down means we need an additional parameter in constructor to specify which kind of FBO we want, and wrap the constructor code with two if-statements, and a check whether it has a texture in the getter function.
Which one do you guys prefer? I am fine with either one.

Reply:
I think current way is fine